graphene_django/converter.py

```diff
@@ -301,7 +301,24 @@ def dynamic_type():
         if not _type:
             return
 
-        return Field(
+        class CustomField(Field):
+            def wrap_resolve(self, parent_resolver):
+                """
+                Implements a custom resolver which go through the `get_node` method to insure that
+                it goes through the `get_queryset` method of the DjangoObjectType.
+                """
+                resolver = super().wrap_resolve(parent_resolver)
+
+                def custom_resolver(root, info, **args):
+                    fk_obj = resolver(root, info, **args)
```

Comment:
This is an issue I've also been getting, queries are forced every time an object is resolved, making an exponential number of queries when nesting without any possibility of optimizing

Reply:
I am fairly new to this code base and I am trying to resolve the issue raised by @ekampf 

In my attempt to fix this:
- I started by getting the id of the field to be resolved (which is the foreign-key object) from the root in an attempt to stop calling one of the functions that resulting in a hit to the database.
- Then I check the resolver
  - No Custom resolver  
    - if the `resolver` equals to the default resolver (meaning that there is no custom resolver for the field in the root node), call `get_node` and return.
    - At this point and this scenario, the issue is solved, because there is one resolver that was called resulting in one hit to the database and thats fine!
  - Custom Resolver Exists
    - At this point, there is a custom resolver, which means that the node should go through multiple resolvers: the `resolver` method + `get_node` (I say should because the tests and the code written to make sure that the processes of resolving the node always goes through the `get_node`)
    - ** In this case, I did not include when the resolver is awaitable

- But the wall I faced after that is the fact that if is required to call `get_node` even though the node is resolved as a result of calling the `resolver` function. Which stopped me for a second, why there are two functions responsible for resolving the 

@tcleonard any thoughts how this can be fixed?

I think this is not the right way to assure that `get_node` is always called. The solution should be more concrete way to do so. 
I was thinking of a way to pass the instance returned from `get_node` to the resolver function if node exists.

Reply:
Thanks for pointing out the issues, and for attempting to fix! I the meantime until a solution can be worked out that doesn't cause N+1 queries and doesn't prevent query-optimization (assuming the original PR author or someone else do care to support the original intention of this PR), I've opened a PR to revert the problematic code here https://github.com/graphql-python/graphene-django/pull/1401, since this caused a significant regression relative to `graphene-django` v3.0.0b7 is a blocker for many people looking to upgrade to graphene-django v3.